File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/StatisticsSample/src/com/hp/workpath/sample/statisticsample/view/faxinfo/FaxInInfoView.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.statisticsample.view.faxinfo;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.hp.workpath.api.statistics.jobinfo.faxinfo.FaxInInfo;
import com.hp.workpath.sample.statisticsample.R;
import com.hp.workpath.sample.statisticsample.view.Utils;

public class FaxInInfoView {

    LinearLayout rootView;
    View view;

    ViewGroup layoutStationId;
    ViewGroup layoutTotalImagesReceived;
    ViewGroup layoutName;
    ViewGroup layoutNumber;

    public FaxInInfoView(LayoutInflater inflater, LinearLayout rootView) {
        this.rootView = rootView;
        this.view = inflater.inflate(R.layout.layout_fax_in_info, rootView, false);
        initViewFaxInInfo();
    }

    public void setFaxInInfo(FaxInInfo faxInInfo) {
        rootView.removeAllViews();
        if (faxInInfo != null) {
            setId(faxInInfo.getCallerId());
            Utils.setSummary(layoutStationId, faxInInfo.getStationId());
            Utils.setSummary(layoutTotalImagesReceived, faxInInfo.getTotalImagesReceived());
            rootView.addView(view);
        } else {
            LinearLayout parent = (LinearLayout) rootView.getParent();
            parent.setVisibility(View.GONE);
        }
    }

    public void setId(FaxInInfo.Id id) {
        if (id != null) {
            Utils.setSummary(layoutName, id.getName());
            Utils.setSummary(layoutNumber, id.getNumber());
        } else {
            LinearLayout parent = (LinearLayout) layoutName.getParent().getParent();
            parent.setVisibility(View.GONE);
        }
    }

    private void initViewFaxInInfo() {
        ((TextView) view.findViewById(R.id.titleCallerIdTextView)).setText(R.string.callerId);
        layoutStationId = Utils.setTitle(view.findViewById(R.id.layoutStationId), R.string.stationId);
        layoutTotalImagesReceived = Utils.setTitle(view.findViewById(R.id.layoutTotalImagesReceived), R.string.totalImagesReceived);
        layoutName = Utils.setTitle(view.findViewById(R.id.layoutName), R.string.name);
        layoutNumber = Utils.setTitle(view.findViewById(R.id.layoutNumber), R.string.number);
    }
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/StatisticsSample/src/com/hp/workpath/sample/statisticsample/view/faxinfo/FaxInInfoView.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.statisticsample.view.faxinfo

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import com.hp.workpath.api.statistics.jobinfo.faxinfo.FaxInInfo
import com.hp.workpath.api.statistics.jobinfo.faxinfo.FaxInInfo.Id
import com.hp.workpath.sample.statisticsample.R
import com.hp.workpath.sample.statisticsample.view.Utils.setSummary
import com.hp.workpath.sample.statisticsample.view.Utils.setTitle

class FaxInInfoView(inflater: LayoutInflater, var rootView: LinearLayout) {
    var view: View = inflater.inflate(R.layout.layout_fax_in_info, rootView, false)

    private lateinit var layoutStationId: ViewGroup
    private lateinit var layoutTotalImagesReceived: ViewGroup

    private lateinit var layoutName: ViewGroup
    private lateinit var layoutNumber: ViewGroup

    fun setFaxInInfo(faxInInfo: FaxInInfo?) {
        rootView.removeAllViews()
        if (faxInInfo != null) {
            setId(faxInInfo.callerId)
            setSummary(layoutStationId, faxInInfo.stationId)
            setSummary(layoutTotalImagesReceived, faxInInfo.totalImagesReceived)
            rootView.addView(view)
        } else {
            val parent = (rootView as ViewGroup).parent as LinearLayout
            parent.visibility = View.GONE
        }
    }

    private fun setId(id: Id?) {
        if (id != null) {
            setSummary(layoutName, id.name)
            setSummary(layoutNumber, id.number)
        } else {
            val parent = layoutName.parent.parent as LinearLayout
            parent.visibility = View.GONE
        }
    }

    private fun initViewFaxInInfo() {
        (view.findViewById<View>(R.id.titleCallerIdTextView) as TextView).setText(R.string.callerId)
        layoutStationId = setTitle(view.findViewById(R.id.layoutStationId), R.string.stationId)
        layoutTotalImagesReceived = setTitle(
            view.findViewById(R.id.layoutTotalImagesReceived),
            R.string.totalImagesReceived
        )
        layoutName = setTitle(view.findViewById(R.id.layoutName), R.string.name)
        layoutNumber = setTitle(view.findViewById(R.id.layoutNumber), R.string.number)
    }

    init {
        initViewFaxInInfo()
    }
}